lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenHoverParticipant.java

```diff
@@ -169,16 +168,31 @@ private String collectPuginConfiguration(IPositionRequest request) {
 		}
 		return null;
 	}
-	
-	private Hover toHover(String description) {
-		Hover hover = new Hover();
-		hover.setContents(new MarkupContent("plaintext", description));
-		return hover;
-	}
 
 	@Override
 	public String onText(IHoverRequest request) throws Exception {
-		// TODO Auto-generated method stub
+		DOMNode tag = request.getNode();
+		String tagText = tag.getNodeValue();
+		if (tagText.startsWith("${" ) && tagText.endsWith("}")) {
```

Comment:
Propertes can be substring, so what you want to check is that the tagText, so you may want to check that request index is *inside* a `${...}` substring

Reply:
So far, the best solution I can come up with for this is to use a regex. Thanks for the requirement :+1:

Reply:
a regexp can work, but another possibility is just to look at last index of `${` (let's call it _indexOpen_) and `}` (_indexCloseBefore_) in the text before the hover location, and first index of `}` (_indexCloseAfter_) after the hover location.
Then `if (indexOpen > indexCloseBefore) propertyName = text.substring(indexOpen, indexCloseAfter);`

Reply:
Okay nice, I'm going to try this solution as having 2 properties in a tagText becomes a bit more complicated with a regexp.